pkg/app/ops/insightcollector/deployments.go

```diff
@@ -451,3 +337,15 @@ func extractChangeFailureRate(deployments []*model.Deployment, from, to int64, t
 		FailureCount: failureCount,
 	}, rest
 }
+
+// groupDeployments groups deployments by applicationID and projectID
+func groupDeployments(deployments []*model.Deployment) (apps, projects map[string][]*model.Deployment) {
```

Comment:
should we add tests for this function 👀

Reply:
Sure.

Reply:
Actually, just found a bug from this function. 👍

Reply:
Added and fixed!